### ISSUE TITLE
Do not require "Name_type" for Names and Descriptions

### ISSUE THREAD

**Issue by jamlung-ri** [OpenConceptLab/ocl_issues#1589]
The [API spec](https://docs.openconceptlab.org/en/latest/oclapi/apireference/concepts.html#create-new-concept) says that concept names and descriptions do not require a Name Type, but the UI does have this field marked as Required (see image).

In case it helps, here's the story: This has created a problem if a user clones and then tries to edit a concept like CIEL's Malaria, which has a name without a name type. The user edits the clone and can't update the clone because CIEL didn't provide a name type. The UI still forces the user to provide one though.

In short, can we make the UI match the API's requirements for concept name types and description types?

![image.png](https://images.zenhubusercontent.com/5df114a3d094b314eeec671c/e1cbd67d-f48a-4a28-96ea-c5e52e548eea)

**Comment by snyaggarwal** [comment]
@paynejd The locale type already has a "Suggested" grouping (from last week).
I have removed the required clause from it in TB.

**Comment by jamlung-ri** [comment]
So it appears that we have two remaining pieces (assuming that the Name Type field is no longer required on the form):

- [ ] Ability to type a custom value in the "Name Type" field if not using OpenMRS validation schema
- [ ] For OpenMRS Validation schema repos, restrict the options to only include the values that OpenMRS allows, with no custom values allowed

**Comment by snyaggarwal** [comment]
@jamlung-ri I have added ability for text entry in name type for non-OpenMRS validation schema Sources.
For OpenMRS validation schema, currently name type that are allowed are -- [FULLY_SPECIFIED, SHORT, INDEX_TERM, None], and thats what the TB also shows. Let me know if there is any change here.

**Comment by snyaggarwal** [comment]
@jamlung-ri this is deployed on QA

**Comment by jamlung-ri** [comment]
This is working well! I think it's ready to deploy.

**Comment by jamlung-ri** [comment]
> Are you saying name type would now be required? This would mean all versions of CIEL (where name type for synonyms is `null`) are now invalid.
Just did some testing, and it is NOT required to have a name type. If you do have a name type, however, it must be in [FULLY_SPECIFIED, SHORT, INDEX_TERM, None]

> Currently, we are using both `null` and `None` for synonyms. The problem is more fully described on [this Talk post](https://talk.openmrs.org/t/handling-name-type-for-synonyms-within-ocl/39707?u=burke). **tl;dr** when we import concepts, we are storing `null` for name type of synonyms; when we edit concepts in OCL, `None` is being stored as the name type for synonyms.
To add to this, as Burke has previously mentioned, the UI in OCL shows "None" (see image) whether the Name Type is null or actually "None". 
![image](https://github.com/OpenConceptLab/ocl_issues/assets/55999682/2d18ef01-e4cf-4a45-9264-437e72655ed8)


> Can we fix the concept editing form so is converts `null` name type to `None` in the UI and then stores any `None` as `null` when the concept is saved (i.e., continue to use `null` for synonyms)?
This is supposed to be specific to OpenMRS Validation Schema users, right? I can imagine a world where the UI shows the term "None" but that's actually equivalent to 'null' being stored in the API. But that might be confusing for users when they go back and forth between sources that do and don't have that OpenMRS schema enabled.

**Comment by jamlung-ri** [comment]
Let's queue this up for an OCL Arch call. 

Main issue for OpenMRS is that the OpenMRS approach has different name types, except for Synonyms. (That is an assumption in OpenMRS that a non-specified name is a synonym.) Any OpenMRS data is `null`, but the form in the TermBrowser uses Name Types that are in OCL, which align with OpenMRS EXCEPT for the way that None is used as a synonym.

There's a difference between how we're using OCL with OpenMRS for None/Synonyms.

**Comment by bmamlin** [comment]
I would propose we deprecate and eliminate (delete & nullify any references to it) the [`None` name type](https://app.openconceptlab.org/#/orgs/OCL/sources/NameTypes/concepts/None/) from OCL. Having both `null` and `None` as possibilities is confusing and is surely going to introduce some problems down the road (if it hasn't already).

If we want to introduce `Synonym` as a name type (instead of `None`), it might be easier for users to understand and we could evolve toward using `Synonym` instead of `null` everywhere (the OCL module for OpenMRS already [understands "Synonym" as a valid name type]([Synonym](https://github.com/openmrs/openmrs-module-openconceptlab/blob/47a90c9ede3da2f6354e1816f27c06b13090ab7c/api/src/main/java/org/openmrs/module/openconceptlab/importer/Saver.java#L281)) and we could easily adapt our import script to label any synonyms as "Synonym" instead of an [empty string](https://github.com/OpenConceptLab/ocl_omrs/blob/master/omrs/management/commands/extract_db.py#L443).

For now, making name types optional in the Term Browser at least allows synonyms to be stored without a name type; however, as long as `None` exists as a name type in OCL, it's likely people will be selecting it.

**Comment by paynejd** [comment]
@jamlung-ri @snyaggarwal Can we close this? The remaining discussion here is about resolving the confusing null/None name type issue, which has moved over to #1666

**Comment by snyaggarwal** [comment]
Yes.
